pydicom/pixel_data_handlers/gdcm_handler.py

```diff
@@ -30,7 +61,16 @@ def get_pixeldata(dicom_dataset):
         raise ImportError(msg)
 
     gdcm_image_reader = gdcm.ImageReader()
-    gdcm_image_reader.SetFileName(dicom_dataset.filename)
+    try:
+        if isinstance(dicom_dataset.filename, unicode):
+            gdcm_image_reader.SetFileName(
```

Comment:
maybe just use `if in_py2 and isinstance...` to avoid the duplicated call and the try/catch